hg-fast-export.sh

```diff
@@ -3,7 +3,7 @@
 # Copyright (c) 2007, 2008 Rocco Rutte <pdmef@gmx.net> and others.
 # License: MIT <http://www.opensource.org/licenses/mit-license.php>
 
-ROOT="$(dirname "$(which "$0")")"
+ROOT="$(dirname "$(greadlink -f "$(which "$0")" 2> /dev/null || readlink -f "$(which "$0")")")"
```

Comment:
Sorry, I‘m no native speaker :-). I like your suggestion using an additional variable for its readability - I‘m going to alter my code accordingly.